content/concepts/secure-comm/tls.md

```diff
@@ -6,5 +6,129 @@ aliases:
+in the first `ClientHello` message by assuming which key agreement algorithm the
+server will choose due to the limited cipher suites. This saves one round trip as the server
+can generate its key from the first message.
+
+As a result, TLS 1.3 uses a new handshake protocol that allows for faster and more efficient
+establishment of encrypted connections. TLS 1.3 also introduces new features, such as
+certificate compression and support for cryptographic keys larger than 4096 bits.
+
+## TLS 1.3 in libp2p
+
+To use TLS 1.3 in libp2p, a peer must first establish a TLS 1.3 connection with another peer
+using the handshake protocol. Once the handshake is complete, the peers can use the encrypted
+connection to exchange data securely and privately.
+
+### Handshake
+
+A TLS 1.3 handshake in libp2p conducts both key sharing and peer authentication
+in a single round trip. Peers must authenticate each other’s identity during the
+handshake. By extension, servers require client authentication during the TLS
+handshake and will abort if the client fails to authenticate.
+
+TLS 1.3 is identified during protocol negotiation with the following protocol
+ID: `/tls/1.0.0`.
+
+In libp2p, endpoints authenticate peers by encoding their public key into an X.509
+certificate extension, however, peers can use arbitrary key types and are not constrained
+to those for which the signing of an X.509 certificate is specified.
+
+{{< alert icon="💡" context="note" text="X.509 is an <a class=\"text-muted\" href=\"https://www.itu.int/en/Pages/default.aspx\"> ITU</a> standard defining the format of public key certificates that use asymmetric cryptography for authentication. Certificate extensions were introduced in version 3 of the X.509 standard, which is a field that offers a set of additional attributes that can be included in the certificate to provide more information about the certificate's subject, such as the certificate's intended purpose, the cryptographic algorithms that the certificate uses, and other relevant details."/>}}
+
+For arbitrary key types, an endpoint needs to encode its host key using the
+[libp2p public key extension](https://github.com/libp2p/specs/blob/master/tls/tls.md#libp2p-public-key-extension)
+which is carried in a self-signed certificate to prove ownership of its host key.
+The libp2p public key extension is an X.509 extension with the Object Identier
+`1.3.6.1.4.1.53594.1.1`. A certificate must include the libp2p public key extension
+to be marked valid.
+
+The endpoint generates a signature using its private host key and shares it along with
+its public host key for verification. The signature proves that the peer owned
+the private host key when signing the certificate.
+
+The public host allows the other peer to calculate the peer ID of the endpoint it
+connects to. Failed authentication would immediately terminate the secure connection
+establishment.
```

Comment:
When processing the certificate, nodes derive the peer ID from the public key that they received,
and clients check that it matches the peer ID of the server they intended to connect to.